utkdigitalinitiatives/islandora_scholar
# Address Trac-746.

**JIRA Ticket**: [TRAC-746](https://jira.lib.utk.edu/browse/TRAC-746)

# What does this Pull Request do?

This PR addresses TRAC-746 by removing previews and the viewer for theses, but not for citations (articles).  In

# What's new?

The utilities.inc file is no longer being used.  Instead, separate includes are created--1 for theses and 1 for citations.  Right now, all of the operations being called from the .module file are now calling the new file, but the old file is still there.  Therefore, we could change this so that only the new stuff calls independent includes.

Some more thinking here:  we're going to have to deal with merge conflicts the more we have to edit stuff to our own liking.  By separating these files out, we can still track head and simply merge what we want in to our new files (limiting conflicts).

# How should this be tested?

If you look at an article, you should see the preview.  If you look at a thesis, you shouldn't.

Also, all other derivative generation and ingestion processes are affected by this change, but should work as normal.

# Additional Notes:

As I said earlier, we are using our own utillities.inc files.  This allows us to more easily track head with limited merge conflicts (just to the .module file).

One issue here is that there is a lot of duplicate code.  We could do the same idea (2 files), but only call the new files for islandora_view_object($object) calls from .module and limit the new files to only be related to this hook.

Rather than merge, let's talk about this and what we may want to do.  We can get fancy here, but it could increase the amount of time needed to complete so let's keep that in mind.

# Interested parties
@CanOfBees @DonRichards @pc37utn @robert-patrick-waltz
